test/TestHelpers.jl

```diff
@@ -198,6 +198,24 @@ end
     val
 end
 
+@inline function Base.deleteat!(A::OffsetArray, i::Int)
+    checkbounds(A, i)
```

Comment:
While not a big deal, changing the bounds checking in the `OffsetArray` code feels a little outside the scope of this PR. As it stands, this has progressed from a 1 word change to partly rewriting `filter!`, adding tests and extending the `OffsetArray`s code.